### PR TITLE
DOC: whatsnew 2.1.0 refinements

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Now there's a fun PR number!

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -130,8 +130,10 @@ Copy-on-Write improvements
 
 .. _whatsnew_210.enhancements.map_na_action:
 
-``map(func, na_action="ignore")`` now works for all array types
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+New :meth:`DataFrame.map` method and support for ExtensionArrays
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :meth:`DataFrame.map` been added and :meth:`DataFrame.applymap` has been deprecated. :meth:`DataFrame.map` has the same functionality as :meth:`DataFrame.applymap`, but the new name better communicates that this is the :class:`DataFrame` version of :meth:`Series.map` (:issue:`52353`).
```
Comment (rhshadrach) [inline doc/source/whatsnew/v2.1.0.rst:136]: It seemed to me we were burying the lead here; wanted to make the additional method to DataFrame be at the top.

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -222,50 +222,51 @@ If the input contains NA values, the previous version would drop those as well w
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
 - :meth:`Series.ffill` and :meth:`Series.bfill` are now supported for objects with :class:`IntervalDtype` (:issue:`54247`)
-- :meth:`Categorical.map` and :meth:`CategoricalIndex.map` now have a ``na_action`` parameter.
-  :meth:`Categorical.map` implicitly had a default value of ``"ignore"`` for ``na_action``. This has formally been deprecated and will be changed to ``None`` in the future.
+- Added ``filters`` parameter to :func:`read_parquet` to filter out data, compatible with both ``engines`` (:issue:`53212`)
```
Comment (rhshadrach) [inline doc/source/whatsnew/v2.1.0.rst:225]: This note was moved from the bugfix section

File: doc/source/whatsnew/v2.1.0.rst [hunk 3]
```diff
@@ -275,6 +276,9 @@ Notable bug fixes
 
 These are bug fixes that might have notable behavior changes.
 
+-
+-
+
```
Comment (rhshadrach) [inline doc/source/whatsnew/v2.1.0.rst:281]: This section is blank; I just wanted to make this more apparent.  cc @lithomas1

File: doc/source/whatsnew/v2.1.0.rst [hunk 4]
```diff
@@ -374,6 +378,8 @@ Deprecations
 Deprecated silent upcasting in setitem-like Series operations
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+https://pandas.pydata.org/pdeps/0006-ban-upcasting.html
```
Comment (rhshadrach) [inline doc/source/whatsnew/v2.1.0.rst:381]: I thought the link to the PDEP could be more prominent.

**Review by rhshadrach — COMMENTED** [review]
@MarcoGorelli - one of the side-effects of the autosort in the whatsnew is that some notes got moved around here making review slightly harder. It looks like it happened only a few times, not a big deal in my opinion. Still very positive on autosort in general.

File: doc/source/reference/series.rst [hunk 5]
```diff
@@ -327,6 +328,7 @@ Datetime properties
    Series.dt.tz
    Series.dt.freq
    Series.dt.unit
+   Series.dt.normalize
```
Comment (rhshadrach) [inline doc/source/reference/series.rst:331]: @mroeschke - this and the one above were added in #52503. They were missing from the API docs. Does this placement look good?

File: doc/source/whatsnew/v2.1.0.rst (hunk 3 above)
Comment (lithomas1) [reply doc/source/whatsnew/v2.1.0.rst:281]: My bad, I deleted the lines but not the section 🤦 .

Can we just take the whole section out?

Comment (rhshadrach) [reply doc/source/whatsnew/v2.1.0.rst:281]: Sounds good! I was thinking we may still add something here, but that seems quite unlikely. Will remove in this PR.

File: doc/source/reference/series.rst (hunk 5 above)
Comment (mroeschke) [reply doc/source/reference/series.rst:331]: Looks good, thanks!

File: doc/source/whatsnew/v2.1.0.rst [hunk 6]
```diff
@@ -44,7 +44,7 @@ This release introduces an option ``future.infer_string`` that infers all
 strings as PyArrow backed strings with dtype ``pd.ArrowDtype(pa.string())`` instead.
 This option only works if PyArrow is installed. PyArrow backed strings have a
 significantly reduced memory footprint and provide a big performance improvement
-compared to NumPy object.
+compared to NumPy object (:issue:`#54430`).
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:47]:
```suggestion
compared to NumPy object (:issue:`54430`).
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 7]
```diff
@@ -222,50 +222,51 @@ If the input contains NA values, the previous version would drop those as well w
...
 - :meth:`DataFrame.applymap` now uses the :meth:`~api.extensions.ExtensionArray.map` method of underlying :class:`api.extensions.ExtensionArray` instances (:issue:`52219`)
 - :meth:`MultiIndex.sort_values` now supports ``na_position`` (:issue:`51612`)
 - :meth:`MultiIndex.sortlevel` and :meth:`Index.sortlevel` gained a new keyword ``na_position`` (:issue:`51612`)
 - :meth:`arrays.DatetimeArray.map`, :meth:`arrays.TimedeltaArray.map` and :meth:`arrays.PeriodArray.map` can now take a ``na_action`` argument (:issue:`51644`)
 - :meth:`arrays.SparseArray.map` now supports ``na_action`` (:issue:`52096`).
-- :meth:`pandas.read_html` now supports the ``storage_options`` keyword when used with a URL, allowing users to add headers the outbound HTTP request (:issue:`49944`)
-- Add :meth:`diff()` and :meth:`round()` for :class:`Index` (:issue:`19708`)
+- :meth:`pandas.read_html` now supports the ``storage_options`` keyword when used with a URL, allowing users to add headers to the outbound HTTP request (:issue:`49944`)
+- Add :meth:`Index.diff()` and :meth:`Index.round()` (:issue:`19708`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:236]:
```suggestion
- Add :meth:`Index.diff` and :meth:`Index.round` (:issue:`19708`)
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 8]
```diff
@@ -864,9 +868,9 @@ Other
...
 - Bug in :meth:`DataFrame.reindex` with a ``fill_value`` that should be inferred with a :class:`ExtensionDtype` incorrectly inferring ``object`` dtype (:issue:`52586`)
-- Bug in :meth:`DataFrame.shift` and :meth:`Series.shift` and :meth:`DataFrameGroupBy.shift` when passing both "freq" and "fill_value" silently ignoring "fill_value" instead of raising ``ValueError`` (:issue:`53832`)
+- Bug in :meth:`DataFrame.shift` and :meth:`Series.shift` and :meth:`.DataFrameGroupBy.shift` when passing both ``freq`` and ``fill_value`` silently ignoring ``fill_value`` instead of raising ``ValueError`` (:issue:`53832`)
 - Bug in :meth:`DataFrame.shift` with ``axis=1`` on a :class:`DataFrame` with a single :class:`ExtensionDtype` column giving incorrect results (:issue:`53832`)
-- Bug in :meth:`GroupBy.first` and :meth:`GroupBy.last` where an empty group would return ``np.nan`` instead of a an ExtensionArray's NA value (:issue:`39098`)
+- Bug in :meth:`DataFrameGroupBy.first`, :meth:`DataFrameGroupBy.last`, :meth:`SeriesGroupBy.first`, and :meth:`SeriesGroupBy.last` where an empty group would return ``np.nan`` instead of the corresponding :class:`.ExtensionArray` NA value (:issue:`39098`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.1.0.rst:873]: needs a "." before DataFrameGroupBy to render

File: doc/source/whatsnew/v2.1.0.rst (hunk 6 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v2.1.0.rst:47]: Thanks!

File: doc/source/whatsnew/v2.1.0.rst (hunk 8 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v2.1.0.rst:873]: Thanks!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**